src/core/socket.c

```diff
@@ -2651,124 +2652,177 @@ static int socket_deserialize_item(Unit *u, const char *key, const char *value,
                         s->control_command = s->exec_command[id];
                 }
         } else if (streq(key, "fifo")) {
-                int fd, skip = 0;
+                _cleanup_free_ char *fdv = NULL;
+                bool found = false;
+                int fd;
 
-                if (sscanf(value, "%i %n", &fd, &skip) < 1 || fd < 0 || !fdset_contains(fds, fd))
+                r = extract_first_word(&value, &fdv, NULL, 0);
+                if (r <= 0) {
                         log_unit_debug(u, "Failed to parse fifo value: %s", value);
-                else {
-                        bool found = false;
-
-                        LIST_FOREACH(port, p, s->ports)
-                                if (p->fd < 0 &&
-                                    p->type == SOCKET_FIFO &&
-                                    path_equal_or_files_same(p->path, value+skip, 0)) {
-                                        p->fd = fdset_remove(fds, fd);
-                                        found = true;
-                                        break;
-                                }
-                        if (!found)
-                                log_unit_debug(u, "No matching fifo socket found: %s", value+skip);
+                        return 0;
                 }
 
+                if (safe_atoi(fdv, &fd) < 0 || fd < 0 || !fdset_contains(fds, fd)) {
```

Comment:
did you mean to switch this to parse_fd()?

Reply:
*sigh*, I did, but for some reason I forgot to commit it